README.md

```diff
@@ -1,14 +1,124 @@
+Both methods uses generics to return a type defined by the default value type.
+
+The method `getFlag` returns an `Evaluation` object that contains the `value` of the flag, the `reason` for the value returned, and the `variant` selected.
+
+In the case of an error, the default value will be returned and the `Evaluation` contains information about the error.
+
+The method `getValue` will simply return the assigned value or the default.
+
+```kotlin
+
+val message: String = confidence.getValue("flag-name.message", "default message")
+val messageFlag: Evaluation<String> = confidence.getFlag("flag-name.message", "default message")
+
+val messageValue = messageFlag.value
+// message and messageValue are the same
+```
+
+### Tracking an event
+Events are defined by a `name` and a `message` where the message is a key-value map of type `<String, ConfidenceValue>`. You can track an event using the `track` method.
+
+All `context` data set on the `Confidence` instance will be appended to the event and its message.
+
+```kotlin
+confidence.track("button-tapped", mapOf("button_id" to ConfidenceValue.String("purchase_button")))
+```
+
+The Confidence SDK has support for `EventProducer`. This is a way to programmatically emit context changes and events into streams 
+which can be consumed by the SDK to automatically emit events or to automatically update context data.
+
+The Confidence SDK comes with a pre-defined event producer to emit some application lifecycle events: `AndroidLifecycleEventProducer`. To use it:
+```kotlin
+import com.spotify.confidence.AndroidLifecycleEventProducer
+confidence.track(
+    AndroidLifecycleEventProducer(
+        application = getApplication(),
+        trackActivities = false // or true
+    )
+)
+```
+
+## Apply events
+This SDK automatically emits `apply` events to the Confidence backend once a flag is accessed. This allows Confidence to track who was exposed to what variant and when.
+
+_Note: the `apply` event is only generated for flags that are successfully evaluated (i.e. default values returned due to errors don't generate `apply` events)._
+_Note: the `apply` event reports which flag and variant was read by the application, but not which property the application has read from such variant's value._
+
+To avoid generating redundant data, as long as the flags' data returned from the backend for a user remains unchanged, only the first time a flag's property is read will generate an `apply` event. This is true also across restarts of the application.
+
+The SDK stores `apply` events on disk until they are emitted correctly, thus ensuring the apply data reaches the backend even if generated when there is no network available (assuming the device will ever re-connect to the network before the application is deleted by the user).
+
+<!-- Add link to the more detailed documentation on apply events in the Confidence portal once it's ready -->
```

Comment:
We have a similar section in the webdocs, perhaps remove this or make it shorter?